### ISSUE TITLE
Database password with special characters causes pg-amqp-bridge to fail

### ISSUE THREAD

**Issue by andrewyager** [FORTH-ICS-INSPIRE/artemis#413]
**Describe the bug**

**Affected Component(s)**
- [X ] Back-End (Database, Microservices, Containers, etc)
- [ ] Front-End (Flask, API, etc)
- [ ] Docs
- [ ] Build System

**To Reproduce**
1. Set a DB_PASSWORD like 123&$FOO
2. Start pg-amqp-bridge
3. observe Illegal character in authority in logs

**Expected behavior**
A. This should be fixed by either documenting that special characters should not be used, or alternatively providing a way to handle this.

We have resolved the issue by adding a new .env variable called DB_PASS_SAFE which has the password percent/url encoded, which Rust can handle. I believe the default config could have DB_PASS_SAFE=${DB_PASS} or similar (although have not tested).

**Comment by slowr** [comment]
Hi @andrewyager , it seems that there are string escape issues and problems with the URIs. Did you also try it with other passwords and had similar issues? I will update the documentation about special characters but I wanted to make sure which ones if not all have issues! 
Cheers

**Comment by andrewyager** [comment]
It seems to be only the URIs that are problematic here. In the default config that is `pg-amqp-bridge`, `graphql` (although this didn't error, so maybe their URI password is more resilient), and `postgrest`. I added an additional .env variable called DB_PASS_SAFE that contained the urlencoded version of the password and modified the container definitions to pull this instead.

I imagine that if I had changed the rabbitmq password from default to something involving a special character, the AMQP URL would also have the issue, however I didn't test that and am just using it as default. I _probably_ should have changed this, but this is hidden behind the firewall for now.
